src-self-hosted/Module.zig

```diff
@@ -851,6 +851,8 @@ pub fn update(self: *Module) !void {
 
     self.generation += 1;
 
+    self.clearErrors();
```

Comment:
Looking at only this single line of code I can already tell you this isn't going to work. Consider this case:

```zig
export fn a() i32 { @compileError("bad"); }
export fn b() i32 { @compileError("bad2"); }
```

expected output:

```
error: 1:12: bad
error: 2:12: bad2
```

Next update:

```zig
export fn a() i32 { return 1234 }
export fn b() i32 { @compileError("bad2"); }
```

expected output:

```
error: 2:12: bad2
```

Now here's the crucial point: `b` is not modified, and nothing that `b` depended on changed. Incremental compilation means that `b` will not need to be re-generated. Its error remains from the previous compilation.